packages/backend-core/src/sql/sql.ts

```diff
@@ -435,6 +435,10 @@ class InternalBuilder {
             )} = ? THEN 1 ELSE 0 END = 1`,
             [value]
           )
+        } else if (this.client === SqlClient.ORACLE) {
+          query = query[fnc](`${quotedIdentifier(this.client, key)} = ?`, [
+            value,
+          ])
```

Comment:
I was the one who originally put the coalesce calls in and they do serve a purpose when handling nulls. Because nulls in SQL tend to propagate, `anything = null` results in `null` rather than a boolean. This then continues to spread throughout the operation, resulting in strange behaviour.

If we were able to test Oracle in our search.spec.ts, I think what we'd find with this change is that it wouldn't be at parity with the other databases. This could be fine in the short term, because it's better than it being flat broken as it currently is, but long term we will want to make this work with coalescing.

Reply:
Makes total sense - I've updated the PR and tests to use an oracle compatible version of `COALESCE`. The problem is that Oracle doesn't like it being used with booleans and it needs a condition at the end. Seems to be working well with the oracle connector after testing too 👍